File: examples/python/self_hosted_client.py
from __future__ import annotations

from cortex.session import MemorySession


def main() -> None:
    session = MemorySession.from_base_url(
        "http://127.0.0.1:8766",
        api_key="replace-me",
        namespace="team",
        actor="examples/python",
    )

    print("sdk:", session.sdk_info())
    print("health:", session.client.health()["status"])

    session.remember(
        label="Project Atlas",
        node_id="atlas",
        brief="Local-first memory runtime",
        aliases=["atlas"],
        tags=["active_priorities"],
        confidence=0.94,
        message="seed atlas from python session example",
    )

    session.remember(
        label="Python SDK",
        node_id="sdk",
        brief="Programmatic Cortex client",
        tags=["infrastructure"],
        confidence=0.88,
        message="seed sdk from python session example",
    )

    session.link(source_id="atlas", target_id="sdk", relation="depends_on")
    context = session.search_context(query="atlas", limit=5)
    branch = session.branch_for_task("Atlas follow-up")

    print("branch:", branch["branch_name"])
    print(context["context"])


if __name__ == "__main__":
    main()

File: examples/typescript/self_hosted_client.mjs
import { MemorySession, SDK_VERSION } from "@cortex-ai/sdk";

async function main() {
  const session = MemorySession.fromBaseUrl("http://127.0.0.1:8766", {
    clientOptions: {
      apiKey: "replace-me",
      namespace: "team"
    },
    sessionOptions: {
      actor: "examples/typescript"
    }
  });

  console.log("sdk", SDK_VERSION, session.sdkInfo());
  console.log("health", (await session.client.health()).status);

  await session.remember({
    label: "Project Atlas",
    nodeId: "atlas",
    brief: "Local-first memory runtime",
    aliases: ["atlas"],
    tags: ["active_priorities"],
    confidence: 0.94,
    message: "seed atlas from typescript session example"
  });

  await session.remember({
    label: "TypeScript SDK",
    nodeId: "ts-sdk",
    brief: "Programmatic Cortex client",
    tags: ["infrastructure"],
    confidence: 0.87,
    message: "seed ts sdk from typescript session example"
  });

  await session.link({
    sourceId: "atlas",
    targetId: "ts-sdk",
    relation: "depends_on"
  });

  const context = await session.searchContext({ query: "atlas", limit: 5 });
  const branch = await session.branchForTask({ task: "Atlas follow-up" });

  console.log("branch", branch.branch_name);
  console.log(context.context);
}

main().catch((error) => {
  console.error(error);
  process.exitCode = 1;
});
